tests/gatewayapitests/npltests/gateway_npl_test.go

```diff
@@ -0,0 +1,453 @@
+/*
+ * Copyright 2023-2024 VMware, Inc.
+ * All Rights Reserved.
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*   http://www.apache.org/licenses/LICENSE-2.0
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package npltests
+
+import (
+	"context"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
+	gatewayfake "sigs.k8s.io/gateway-api/pkg/client/clientset/versioned/fake"
+
+	akogatewayapik8s "github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/ako-gateway-api/k8s"
+	akogatewayapilib "github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/ako-gateway-api/lib"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/k8s"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/lib"
+	avinodes "github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/nodes"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/objects"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pkg/utils"
+	tests "github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/gatewayapitests"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tests/integrationtest"
+)
+
+var ctrl *akogatewayapik8s.GatewayController
+
+const (
+	DEFAULT_NAMESPACE = "default"
+)
+
+func TestMain(m *testing.M) {
+
+	tests.KubeClient = k8sfake.NewSimpleClientset()
+	tests.GatewayClient = gatewayfake.NewSimpleClientset()
+	integrationtest.KubeClient = tests.KubeClient
+
+	// Sets the environment variables
+	os.Setenv("CLUSTER_NAME", "cluster")
+	os.Setenv("CLOUD_NAME", "CLOUD_VCENTER")
+	os.Setenv("SEG_NAME", "Default-Group")
+	os.Setenv("POD_NAMESPACE", utils.AKO_DEFAULT_NS)
+	os.Setenv("FULL_SYNC_INTERVAL", utils.AKO_DEFAULT_NS)
+	os.Setenv("ENABLE_EVH", "true")
+	os.Setenv("TENANT", "admin")
+	os.Setenv("POD_NAME", "ako-0")
+	os.Setenv("SERVICE_TYPE", "NodePortLocal")
+
+	// Set the user with prefix
+	_ = lib.AKOControlConfig()
```

Comment:
We can move statement at Line 70 in place of 66. Use that variable in remaining places like line number 69, 71 etc.

Reply:
line 66 and line 70 refer to different AKOControlConfig objects for ako and gateway-api respectively